### DISCUSSION TITLE
Issue when rendering a disabled field after resetting

### DISCUSSION THREAD

**Discussion by iKrushYou** [react-hook-form/react-hook-form#5365] [Q&A] [answered]
https://codesandbox.io/s/react-hook-form-disabled-field-issue-iwrn0

This is another edgecase i've managed to find. It seems that if I try to render a form with disabled inputs and `reset` the form with the already submitting information it will run into issues. Follow instructions on the CSB

**Comment by bluebill1049** [comment]
can you try with the latest? i think @leapful has fixed this issue.

**Reply by iKrushYou** [reply]
updated the CSB to 7.6.8 but it looks like the same issue is happening. 

Reset will still remove the value

I added a checkbox to enable / disable the fields and it can be tested by
1. hit reset
2. hit next, watch the field appear, but the value stays
3. check "disabled"
4. hit next again, observe field value disappears

**Reply by bluebill1049** [reply]
I haven't got time to look into more detail, but the watch is going to one step behind the disabled input.

watch -> return value
render -> disabled whatever (this value is going to be one tick behind)

**Reply by iKrushYou** [reply]
Hey @bluebill1049 I don't think that sounds like what i'm observing. If you get a chance eventually, check out the demo and let me know

Thanks!

**Comment by leapful** [comment]
@iKrushYou 

It seems you are mis-understanding the "ref" handling of react-hook-form.

In your CSB sample, when "disabled" checkbox is true, the component will perform a re-rendering, at that time, "watch" method will return the current not disabled form values (it's not changed yet to disabled). The new disabled state of form element only be applied after the component finished re-rendering. Since there is not change on form element value, there is no more re-rendering, that's why the last not disabled form values still shows.

The following CSB sample adds a "useEffect" handling to show "watch()" values after disabled state of form element is re-rendered successfully might help you understand this case better:

https://codesandbox.io/s/react-hook-form-disabled-field-issue-forked-ekybi?file=/src/App.tsx

**Reply by iKrushYou** [reply]
Hey @leapful thank you for the response! I can kind of see what you're saying here. Is there a recommended solution to this, though? Or is this expected behavior (doesn't seem like it should be)?

**Reply by leapful** [reply]
@iKrushYou 

Please describe more detail why you need to get form values at re-rendering time of enabled/disabled field element ? The solution will depend on your purpose.

**Reply by bluebill1049** [reply]
thank you for looking into this @leapful 🙏

**Reply by iKrushYou** [reply]
Well, i went to go make an example because I ran into the issue again but then I realized that this was solved in a later version

https://codesandbox.io/s/romantic-payne-00il4?file=/src/App.js

**Comment by iKrushYou** [comment, marked as answer]
Well, i went to go make an example because I ran into the issue again but then I realized that this was solved in a later version

https://codesandbox.io/s/romantic-payne-00il4?file=/src/App.js

You can experiment by changing the version of RHF to latest

**Reply by sebastienbarre** [reply]
I feel like I'm having a similar problem as yours, `watch()` keeps return "undefined" when I set the `disabled` prop of my input fields. When you say "latest" version, do you 7.12, or the development/master version? Thx

**Reply by iKrushYou** [reply]
yeah, I mean `7.12.0`

**Reply by sebastienbarre** [reply]
Thx. In that case, different problem, ugh.

**Reply by iKrushYou** [reply]
Ah, do you have a different discussion outlining your issue?

**Reply by sebastienbarre** [reply]
I googled far and wide,  looked at a lot of issues in the tracker, and couldn't find anything similar, so my gut feeling is that I'm misunderstanding how it works. I believe @bluebill1049 made it clear that this library tries to be as close as possible to HTML standards, so disabled inputs will *not* be submitted, and that's understandable. But I could have sworn I read that you could still *read* the values of disabled input using `watch` or `getValues`, since I assume those are kept around as a state object internally, right? But it doesn't appear to be the case. Thx. Sorry, didn't mean to hijack the issue selected answer.

**Reply by bluebill1049** [reply]
@sebastienbarre we will introduce a new `disabled` prop for `register`. The current `disabled` prop is suitable for static `disabled`, hopefully, this problem will get improved once that new prop gets introduced.
